sympy/functions/elementary/hyperbolic.py

```diff
@@ -1693,6 +1693,15 @@ def expansion_term(n, x, *previous_terms):
                 F = factorial(k) * n // 2 * n // 2
                 return -1 * R / F * x**n / 4
 
+    def _eval_as_leading_term(self, x, logx=None, cdir=0):
+        from sympy.series.order import Order
+        arg = self.args[0].as_leading_term(x)
+
+        if x in arg.free_symbols and Order(1, x).contains(arg):
```

Comment:
`contains` is an expensive method. This check can simply be - 
```py
        arg0 = arg.subs(x, 0)
        if arg0 is S.NaN:
            arg0 = arg.limit(x, 0, dir='-' if cdir.is_negative else '+')
        if arg0.is_zero:
            return -1*log(arg)
```

Reply:
Yeah this had crossed this mind actually but then all `trig`, `inverse trig`, `hyperbolic` , `inverse hyperbolic ` use `.contains` for their respective `_eval_as_leading_term` method, hence I thought of going forward with contains for consistency . ! Though won't be an issue if the change is introduced everywhere !! I'll try that out and get back

Reply:
I have removed the `contains` method from all `hyperbolic` and `inverse hyperbolic` functions of `hyperbolic.py` as this block shown above is more than sufficient for all functions and should perform better than using than `contains`.